mono/utils/mono-tls.h

```diff
@@ -101,18 +110,190 @@ mono_native_tls_set_value (MonoNativeTlsKey key, gpointer value)
 void mono_tls_init_gc_keys (void);
 void mono_tls_init_runtime_keys (void);
 void mono_tls_free_keys (void);
-gint32 mono_tls_get_tls_offset (MonoTlsKey key);
-
-G_EXTERN_C MonoInternalThread *mono_tls_get_thread (void);
-G_EXTERN_C MonoJitTlsData     *mono_tls_get_jit_tls (void);
-G_EXTERN_C MonoDomain *mono_tls_get_domain (void);
-G_EXTERN_C SgenThreadInfo     *mono_tls_get_sgen_thread_info (void);
-G_EXTERN_C MonoLMF           **mono_tls_get_lmf_addr (void);
-
-void mono_tls_set_thread 	   (MonoInternalThread *value);
-void mono_tls_set_jit_tls 	   (MonoJitTlsData     *value);
-void mono_tls_set_domain 	   (MonoDomain         *value);
-void mono_tls_set_sgen_thread_info (SgenThreadInfo     *value);
-void mono_tls_set_lmf_addr 	   (MonoLMF           **value);
+
+G_EXTERN_C MonoInternalThread *mono_tls_get_thread_extern (void);
+G_EXTERN_C MonoJitTlsData     *mono_tls_get_jit_tls_extern (void);
+G_EXTERN_C MonoDomain *mono_tls_get_domain_extern (void);
+G_EXTERN_C SgenThreadInfo *mono_tls_get_sgen_thread_info_extern (void);
+G_EXTERN_C MonoLMF       **mono_tls_get_lmf_addr_extern (void);
+
+/*
+ * On all platforms we should be able to use either __thread or __declspec (thread)
```

Comment:
Did you mean to also leave the original copy of these paragraphs in `mono-tls.c`

Reply:
I meant to move almost everything to the .h.
I kinda didn't think about where is most readable.
And larger .h is bad for compile perf perhaps, at least it used to be.
I can look at it again and try to decide what is most readable.

I'd really like to remove the non-default `__thread` stuff, but I kept it.